src/main/java/com/google/sps/data/GivrUser.java

```diff
@@ -104,6 +106,29 @@ public static void updateUserInDatastore(String identifyingProperty, String iden
     datastore.put(entity);
   }
 
+  public void setModeratingOrgs() {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    this.moderatingOrgs = new ArrayList<Entity>();
+
+    Query query = new Query("Distributor").addSort("creationTimeStampMillis", SortDirection.DESCENDING);
+
+    // TODO(): Check in invitedModeratorList as well.
+
+    Filter filter = null;
+    try {
+      filter = new FilterPredicate("moderatorList", FilterOperator.IN, this.email);
```

Comment:
shouldn't you be looking for user id's and not emails since moderator list contains their ids? Also what happens if they are an invited moderator but not in the moderator list yet

Reply:
fixed by using Composite Filters!